dotcom-rendering/src/web/components/StickyVideo.importable.tsx

```diff
@@ -91,27 +92,64 @@ const stickyContainerStyles = (height: number) => css`
 interface Props {
 	isPlaying: boolean;
 	children: React.ReactNode;
+	videoId: string;
 }
 
-export const StickyVideo = ({ isPlaying, children }: Props) => {
+export const StickyVideo = ({ isPlaying, children, videoId }: Props) => {
 	const [isSticky, setIsSticky] = useState(false);
+	const [isClosed, setIsClosed] = useState(false);
+
 	const [isIntersecting, setRef] = useIsInView({
 		threshold: 0.5,
 		repeat: true,
 	});
 
 	useEffect(() => {
 		setIsSticky(isPlaying && !isIntersecting);
+		setIsClosed(false);
 	}, [isIntersecting, isPlaying]);
 
+	useEffect(() => {
+		if (isSticky) {
+			submitComponentEvent({
+				component: {
+					componentType: 'STICKY_VIDEO',
+					id: videoId,
+				},
+				action: 'STICK',
+			});
+		} else if (isClosed) {
+			submitComponentEvent({
```

Comment:
Does this need to live in the useEffect? I think I'd imagined we'd just trigger the event when the button was clicked, maybe I'm missing something though - is there a reason to put it in here?